satpy/modifiers/angles.py

```diff
@@ -274,8 +275,20 @@ def get_satellite_zenith_angle(data_arr: xr.DataArray) -> xr.DataArray:
     return satz
 
 
+def get_cos_sza(data_arr: xr.DataArray) -> xr.DataArray:
+    """Generate the cosine of the solar zenith angle for the provided data.
+
+    Returns:
+        DataArray with the same shape as ``data_arr``. Units are radians.
```

Comment:
Cosine of something is not in any angle units. It is the ratio of the length of cathet opposite to the angle and the length of the hypotenuse.

Reply:
:man_facepalming: Done. Thanks.